src/gui/log/loglistview.cpp

```diff
@@ -0,0 +1,140 @@
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "loglistview.h"
+
+#include <QApplication>
+#include <QClipboard>
+#include <QFontMetrics>
+#include <QKeyEvent>
+#include <QPainter>
+#include <QStyle>
+#include <QStyledItemDelegate>
+
+#include "logmodel.h"
+#include "uithememanager.h"
+
+namespace
+{
+    const QString SEPARATOR = QStringLiteral(" - ");
+
+    int horizontalAdvance(const QFontMetrics &fontMetrics, const QString &text)
+    {
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 11, 0))
+        return fontMetrics.horizontalAdvance(text);
+#else
+        return fontMetrics.width(text);
+#endif
+    }
+
+    QString logText(const QModelIndex &index)
+    {
+        return QString::fromLatin1("%1%2%3").arg(index.data(BaseLogModel::TimeRole).toString(), SEPARATOR
+                                                 , index.data(BaseLogModel::MessageRole).toString());
+    }
+
+    class LogItemDelegate : public QStyledItemDelegate
+    {
+    public:
+        using QStyledItemDelegate::QStyledItemDelegate;
+
+    private:
+        void paint(QPainter *painter, const QStyleOptionViewItem &option, const QModelIndex &index) const override
+        {
+            painter->save();
+            QStyledItemDelegate::paint(painter, option, index); // paints background, focus rect and selection rect
+
+            const QStyle *style = option.widget ? option.widget->style() : QApplication::style();;
+            const QRect textRect = style->subElementRect(QStyle::SE_ItemViewItemText, &option, option.widget)
+                                   .adjusted(1, 0, 0, 0); // shift 1 to avoid text being too close to focus rect
+
+            QFont font = option.font;
+            if (option.font.pointSize() > 0)
+                font.setPointSize(option.font.pointSize()); // somehow this needs to be set directly otherwise painter will use default font
```

Comment:
It still seems weird... if copy constructor doesn't produce the same "value" there is bug in it.

Reply:
I couldn't get if I was doing something fishy, everything is const, so I don't think I can do anything here then
I tried checking qt sources for qfont but they are using private impl so I didn't try to further delve into it.

Reply:
I tried sanitizer, both master and this branch seems to be throwing same warnings (there were quite a number of warnings especially with thread sanitizer)

Reply:
I wonder how QStyledItemDelegate draws the text?

Reply:
But it's most likely a problem with QFont since an action that shouldn't have any effect still has it.
Just for fun... can you test the following?
```c++
font.setPointSize(font.pointSize());
```

Reply:
hmm it works 
![image](https://user-images.githubusercontent.com/34717789/79464484-bce5c780-8017-11ea-9733-83405299b002.png)

Q_ASSERT(font.pointSize() != option.font.pointSize());
actually fails